example/merge_intervals.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2023 Jiri Nytra (jiri.nytra at gmail dot com)
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <flux.hpp>
+
+#include <cstddef>
+#include <iostream>
+#include <vector>
+
+struct interval_t
+{
+    std::size_t begin;
+    std::size_t end;
+};
+
+std::ostream& operator<<(std::ostream& s, const interval_t& i)
+{
+    return s << '(' << i.begin << ',' << i.end << ')';
+}
+
+bool is_overlapped(interval_t a, interval_t b)
+{
+    return a.end >= b.begin;
+}
+
+auto merge = [](auto seq) -> interval_t
+{
+    auto begin = flux::front(seq)->begin;
+    auto end = flux::from(seq).max(flux::proj(std::less<>{}, &interval_t::end))->end;
```

Comment:
It's slightly less typing to use the free-function form of `max` here, i.e.

    auto end = flux::max(seq, flux::proj(std::less<>{}, &interval_t::end))->end;